__tests__/integration/send-tx.test.js

```diff
@@ -1573,4 +1573,109 @@ describe('filter query + custom tokens', () => {
     expect(addr5htr.total_amount_available).toBe(1000);
     done();
   });
+
+  // Outputs must have address and value or type and data
+  it('should reject an invalid output object', async done => {
+    // Output with address and without value
+    const response = await TestUtils.request
+      .post('/wallet/send-tx')
+      .send({
+        outputs: [{
+          address: 'invalidAddress',
+        }],
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    expect(response.status).toBe(400);
+
+    // Output with type 'data' but without data
+    const response2 = await TestUtils.request
+      .post('/wallet/send-tx')
+      .send({
+        outputs: [{
+          type: 'data',
+        }],
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    expect(response2.status).toBe(400);
+    done();
+  });
+});
+
+describe('transaction with data script output', () => {
+  let wallet1; // Receives funds
+  let wallet2; // Main destination for test transactions
+
+  beforeAll(async () => {
+    try {
+      wallet1 = WalletHelper.getPrecalculatedWallet('send-tx-data-output-1');
+      wallet2 = WalletHelper.getPrecalculatedWallet('send-tx-data-output-2');
+
+      await WalletHelper.startMultipleWalletsForTest([wallet1, wallet2]);
+
+      // Funds
+      await wallet1.injectFunds(1000, 0);
+    } catch (err) {
+      TestUtils.logError(err.stack);
+    }
+  });
+
+  afterAll(async () => {
+    await wallet1.stop();
+    await wallet2.stop();
+  });
+
+  it('should success with an output data script', async done => {
+    const tx = await wallet1.sendTx({
+      outputs: [{
+        type: 'data',
+        data: 'test'
+      }],
+    });
+
+    expect(tx.success).toBe(true);
+    expect(tx.hash).toBeDefined();
+
+    // Checking wallet balance
+    // the data script created spent 0.01 HTR
+    const balance = await wallet1.getBalance();
+    expect(balance.available).toBe(999);
+
+    expect(tx.outputs.length).toBe(2);
```

Comment:
The transaction returned in the success does not have any detail to identify the output type. I've done the value check here: https://github.com/HathorNetwork/hathor-wallet-headless/pull/192/commits/b427e373a5e4ad8e19ce27030490e3652cc40e54 and explained it

Reply:
Looks good to me.
Maybe this suggestion I made above would be a better fit to the `__tests__/integration/transaction.test.js` test suite: make a `data` type tx and validate the way `[GET] /wallet/transaction` will show its results.

Do you agree it would be a good idea to add this test there?

Reply:
The transaction API return has the outputs in the following format:

```
"outputs": [
        {
            "value": 1,
            "token_data": 0,
            "script": "BHRlc3Ss",
            "decoded": {},
            "token": "00",
            "spent_by": null,
            "selected_as_input": false
        },
        {
            "value": 1,
            "token_data": 0,
            "script": "BXRlc3QyrA==",
            "decoded": {},
            "token": "00",
            "spent_by": null,
            "selected_as_input": false
        },
        {
            "value": 1,
            "token_data": 0,
            "script": "BHRlc3Ss",
            "decoded": {},
            "token": "00",
            "spent_by": null,
            "selected_as_input": false
        },
        {
            "value": 3,
            "token_data": 0,
            "script": "dqkUADBytzC7857cjVhGmjdzfRUTA6KIrA==",
            "decoded": {
                "type": "P2PKH",
                "address": "WNh2vAJfZm8oBFykZbt6YxiyJGVgGsqczm",
                "timelock": null
            },
            "token": "00",
            "spent_by": null,
            "selected_as_input": false
        },
        {
            "value": 975,
            "token_data": 0,
            "script": "dqkU2IB7zR0Y2H/bJ7XCq2JPY725CliIrA==",
            "decoded": {
                "type": "P2PKH",
                "address": "WiQnkZS6PCw8qSgqid1stTpgNXfZLPwSHM",
                "timelock": null
            },
            "token": "00",
            "spent_by": null,
            "selected_as_input": false
        }
    ],
```

I've created an issue to handle this in the future: https://github.com/HathorNetwork/hathor-wallet-headless/issues/193

For now I guess we could just merge like this, what you think?